### PR TITLE
[MM-10430] Add selector to check if the channel is hidden

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Add selector to check if the channel is hidden

(Didn't added a test as I'm not sure how to effectively test it. Will try to revisit next time)

Server PR - https://github.com/mattermost/mattermost-server/pull/8751
Webapp PR - https://github.com/mattermost/mattermost-webapp/pull/1194

#### Ticket Link
Jira ticket: [MM-10430](https://mattermost.atlassian.net/browse/MM-10430)

### CODE REVIEW

**Comment by saturninoabril** [comment]
Update: While working on Mobile RN, I realized that there are some store related implementation in webapp that is needed by mobile RN as well. This update moves the "hiddenDefaultChannelId" here which was initially introduced at webapp.

@lieut-data Sorry but you might want to take a look again.

**Comment by lieut-data** [comment]
@saturninoabril, will do! Sometimes I miss comments, though, so feel free to always re-add me as a reviewer to put it in my https://github.com/pulls/review-requested list :)

File: test/actions/channels.test.js [hunk 1]
```diff
@@ -1833,4 +1833,21 @@ describe('Actions.Channels', () => {
...
+        let state = store.getState();
+        let hiddenChannel = state.entities.channels.hiddenDefaultChannelId;
+        assert.ok(hiddenChannel);
+        assert.ok(hiddenChannel[teamId] === channelId);
+
+        Actions.removeHiddenDefaultChannel(teamId, channelId)(store.dispatch, store.getState);
+        state = store.getState();
+        hiddenChannel = state.entities.channels.hiddenDefaultChannelId;
+        assert.ok(hiddenChannel);
+        assert.ok(!hiddenChannel[teamId]);
```
Comment (lieut-data) [inline test/actions/channels.test.js]: Worth adding test cases for when the `teamId` or `channelId` is `falsy` when passed to the action creators?

File: src/reducers/entities/channels.js [hunk 2]
```diff
@@ -382,6 +382,27 @@ function stats(state = {}, action) {
     }
 }
 
+function hiddenDefaultChannelId(state = {}, action) {
```
Comment (lieut-data) [inline src/reducers/entities/channels.js]: Worth adding a test case for this reducer as well?

**Comment by saturninoabril** [comment]
Thanks @hmhealey and @lieut-data!
Sorry for not giving update on this PR which I'm still in the process of updating.  Actually you (@hmhealey) and @enahum have the same concern about storing that ID. 
Will just ping both of you whenever this PR is ready for dev review :)

**Comment by saturninoabril** [comment]
@lieut-data @hmhealey I've updated the PR and just retain the selector `shouldHideDefaultChannel`. I initially add other selectors before when I was trying to figure out how to make the default channel become available at "More Channels" modal but now that it's not needed per ticket, I've removed all of those.

**Pull request merged**